engine/access/ingestion2/latest_persisted_sealed_result.go

```diff
@@ -0,0 +1,112 @@
+package ingestion2
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// LatestPersistedSealedResult tracks the most recently persisted sealed execution result processed
+// by the ingestion engine.
+type LatestPersistedSealedResult struct {
+	// resultID is the execution result ID of the most recently persisted sealed result.
+	resultID flow.Identifier
+
+	// height is the height of the most recently persisted sealed result's block.
+	// This is the value stored in the consumer progress index.
+	height uint64
+
+	// cp is the consumer progress instance
+	cp storage.ConsumerProgress
+
+	// writeMu is used to prevent concurrent batch updates to the persisted height.
+	// the critical section is fairly large, so use a separate mutex from the cached values.
+	writeMu sync.Mutex
+
+	// mu is used to protect access to resultID and height.
+	mu sync.RWMutex
+}
+
+// NewLatestPersistedSealedResult creates a new LatestPersistedSealedResult instance.
+// It initializes the consumer progress index using the provided initializer and initial height.
+//
+// No errors are expected during normal operation,
+func NewLatestPersistedSealedResult(
+	initialHeight uint64,
+	initializer storage.ConsumerProgressInitializer,
+	headers storage.Headers,
+	results storage.ExecutionResults,
+) (*LatestPersistedSealedResult, error) {
+	// initialize the consumer progress, and set the initial height if this is the first run
+	cp, err := initializer.Initialize(initialHeight)
+	if err != nil {
+		return nil, fmt.Errorf("could not initialize progress initializer: %w", err)
+	}
+
+	// get the actual height stored
+	height, err := cp.ProcessedIndex()
+	if err != nil {
+		return nil, fmt.Errorf("could not get processed index: %w", err)
+	}
+
+	// finally, lookup the sealed resultID for the height
+	header, err := headers.ByHeight(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get header: %w", err)
+	}
+
+	result, err := results.ByBlockID(header.ID())
```

Comment:
Could we refer to some code that this method returns sealed result or some module that index result after block is sealed?

Reply:
addressed in [6dd46e9](https://github.com/onflow/flow-go/pull/7460/commits/6dd46e93f5083ed11f30e9e75a6134cfe79c7313)